aiida/manage/caching.py

```diff
@@ -62,32 +62,19 @@ def _get_config(config_file):
     # Validate configuration
     for key in config:
         if key not in DEFAULT_CONFIG:
-            raise ValueError("Configuration error: Invalid key '{}' in cache_config.yml".format(key))
+            raise exceptions.ConfigurationError("Configuration error: Invalid key '{}' in cache_config.yml".format(key))
 
     # Add defaults where key is either completely missing or specifies no values in which case it will be `None`
     for key, default_config in DEFAULT_CONFIG.items():
         if key not in config or config[key] is None:
             config[key] = default_config
 
-    # Validate the entry point identifiers
-    for key in [ConfigKeys.ENABLED.value, ConfigKeys.DISABLED.value]:
-
-        # If the key is defined in the file but contains no values, it will be `None`
-        if config[key] is None:
-            continue
-
-        for identifier in config[key]:
-            if not is_valid_entry_point_string(identifier):
-                raise exceptions.ConfigurationError(
-                    "entry point '{}' in 'cache_config.yml' is not a valid entry point string.".format(identifier)
-                )
-
-            try:
-                load_entry_point_from_string(identifier)
-            except exceptions.EntryPointError as exception:
-                raise exceptions.ConfigurationError(
-                    "entry point '{}' in 'cache_config.yml' can not be loaded: {}.".format(identifier, exception)
-                )
+    try:
+        type_check(config[ConfigKeys.DEFAULT.value], bool)
+        type_check(config[ConfigKeys.ENABLED.value], list)
+        type_check(config[ConfigKeys.DISABLED.value], list)
```

Comment:
Yes, I agree it would be good to have at least the validation that it _could_ be a valid process type string. Because of the possible wildcards, we probably have to implement the check separately and can't re-use the ``is_valid_entry_point_string``.

For the entry points, as far as I see the valid form is

``<group_name>:<something>``, where

- ``group_name`` is one of ``aiida.data``, ``aiida.calculations``, ..., as defined by the ``entry_point_group_to_module_path_map``
- ``something`` can be literally anything except it can _not_ contain another semicolon

For the fully qualified python name, each colon-separated part must fulfill
```python
name.isidentifier() and not keyword.iskeyword(name)
```
The only two restrictions I'm aware of (compared to the simple regex) is that it can't start with a number, and can't be a keyword (like ``in``, ``for``, ``else``, ...).

All this is a bit tricky to check for due to the wildcards, but I think I can come up with a reasonable solution.

Reply:
Alright, had a crack at this in `_validate_identifier_pattern`.